lib/blueprinter/extractors/association_extractor.rb

```diff
@@ -7,14 +7,14 @@ def initialize
       @extractor = Blueprinter.configuration.extractor_default.new
     end
 
-    def extract(association_name, object, local_options, options={})
-      options_without_default = options.reject { |k,_| k == :default || k == :default_if }
+    def extract(association_name, object, local_options, options = {})
+      options_without_default = options.reject { |k, _| k == :default || k == :default_if }
       # Merge in assocation options hash
       local_options = local_options.merge(options[:options]) if options[:options].is_a?(Hash)
       value = @extractor.extract(association_name, object, local_options, options_without_default)
       return default_value(options) if use_default_value?(value, options[:default_if])
       view = options[:view] || :default
-      blueprint = association_blueprint(options[:blueprint], value)
+      blueprint = association_blueprint(options[:blueprint], value, association_name)
```

Comment:
Instead of performing the blueprint validation checks here, would you instead perform the validation checks here in `association` https://github.com/procore/blueprinter/blame/eb46aef170b29a3df29d741fc249713382cff0c4/lib/blueprinter/base.rb#L159 instead?

The reasoning is that there is already a check there, but it's not as thorough as what you have here. Let's add your checks there and validate it all in one place. Additionally, the `Base.association`  is happening during initialize time, where as the operation here is happening during runtime. During initialize time, blueprinter is mapping all the blueprints and field names, while during runtime it extracts field values. It's better to do the class validation during initialize time since we want to limit operations in runtime for performance reason.

Reply:
Sure, makes perfect sense. I hadn't noticed that there were some validations in place already. I did what you suggested and also moved [this validation](https://github.com/procore/blueprinter/blame/eb46aef170b29a3df29d741fc249713382cff0c4/lib/blueprinter/base.rb#L159) into the `base_helper.rb`.